src/system-script-setup.js

```diff
@@ -0,0 +1,21 @@
+window.__systemAmdScript = {};
+
+let scriptNameMap = window.__systemAmdScript.scriptNameMap = {};
+let nameCounter = 0;
```

Comment:
If this is going to be a global variable, should we namespace it to the `window.__systemAmdScript` variable?